### ISSUE TITLE
Bug cleanup in Run by Line

### ISSUE THREAD

**Issue by DavidKutu** [microsoft/vscode-jupyter#7218]
High priority

- [x] context keys are sometimes stuck in the "running" state
  - Seems that we run the cell but don't hit the breakpoint, and don't reset the keys. See logs below
- [x] RBL button doesn't show up sometimes
- [x] Sometimes I am blocked by an error with starting the kernel. I get a modal error about a missing debug adapter, and an infinite "Starting run by line" notification
- [x] When stepping over a call to another file, the other file is still opened
   - don't send stopped events to vscode when we're outside the notebook. Check stacktrace request to see if we get response right away.

Lower priority

- [x] show the "starting run by line" notification only the first time
- [x] set global context keys based on active editor, so I can run two RBL sessions in two notebooks
- [x] Context keys are reset when any other debug session ends
   - debug.onDidTerminateDebugSession, check for our own sessions
- [x] setting for optionally showing variable explorer

**Comment by roblourens** [comment]
[worked.log](https://github.com/microsoft/vscode-jupyter/files/7024276/worked.log)
[didntwork.log](https://github.com/microsoft/vscode-jupyter/files/7024277/didntwork.log)

**Comment by DavidKutu** [comment]
add setting for bringing up the variable explorer

**Comment by DavidKutu** [comment]
don't send stopped events to vscode when we're outside the notebook. Check stacktrace request to see if we get response right away.
